hack/lib/util.sh

```diff
@@ -457,7 +457,7 @@ kube::util::godep_restored() {
 kube::util::ensure_clean_working_dir() {
   if ! git diff --exit-code; then
     echo -e "\nUnexpected dirty working directory."
-    exit 1
+    return 1
```

Comment:
You might need to update all the places that use this function, since from what I see they expect `exit 1`. See `hack/update-staging-client-go.sh` `hack/update-staging-godeps.sh` and `hack/update-all-staging.sh`